src/sdk/product/useProductLink.ts

```diff
@@ -1,18 +1,50 @@
-import type { ProductSummary_ProductFragment } from '@generated/graphql'
-import type { CurrencyCode, ViewItemData, ViewItemEvent } from '@faststore/sdk'
-import { useSession, sendAnalyticsEvent } from '@faststore/sdk'
 import { useMemo } from 'react'
+import { useSession, sendAnalyticsEvent } from '@faststore/sdk'
+import type { ProductSummary_ProductFragment } from '@generated/graphql'
+import type {
+  CurrencyCode,
+  ViewItemData,
+  ViewItemEvent,
+  SelectItemEvent,
+  SelectItemData,
+  Item,
+} from '@faststore/sdk'
 
 export type ProductLinkOptions = {
-  slug: string
+  index: number
```

Comment:
I think this API should look like more like this:

```suggestion
  index: number
  product: ProductSummary_ProductFragment
```

It's not a very common practice to use spread operators (`...`) for props/parameters that you actually know the type of.